acados/sim/sim_irk_integrator.h

```diff
@@ -71,6 +71,7 @@ typedef struct
     external_function_generic *nls_y_fun;  // evaluation nls function
     external_function_generic *conl_cost_fun_jac_hess;
     external_function_generic *conl_cost_fun;
+    int outer_hess_is_diag;
```

Comment:
Can we also use a pointer for this, move it to memory, and do it as for `W_chol`?
Then, when doing the change I suggested for NLS, it would automatically update the flag in the integrator.
Moreover, we would not have to set `outer_hess_is_diag` in the dynamics model in the templates at all.

Reply:
addressed in [db8b17f](https://github.com/acados/acados/pull/1024/commits/db8b17f92bcefbea96ba18b3c23524010e80b8dc)